### PR TITLE
CH-277: All alerts page as per the new design

### PR DESCRIPTION
**Pull request by NareshGopal**
# Problem Statement
Modify the "All Alerts" page as per the [new design](https://www.figma.com/file/HuYNdGvVazilCkD8mJO1qQ/LR-Alert-page-redesign-options?node-id=1094%3A68101)

# Design
[Design link](https://www.figma.com/file/HuYNdGvVazilCkD8mJO1qQ/LR-Alert-page-redesign-options?node-id=1094%3A68101)

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Review by sujit-h-singh — CHANGES_REQUESTED** [review]

File: manager/gui/src/components/alerts/LrAlert.constant.ts [hunk 1]
```diff
@@ -87,16 +87,52 @@ export const lrAlertConst = {
...
+            tooltip: "Comment",
+          },
+          {
+            name: "occurrences",
+            path: "occurrences-icon.svg",
+            hoverPath: "occurrences-hover-icon.svg",
+            tooltip: "Occurrences",
+          },
+          {
+            name: "alertDetails",
```
[the remark this answers is not in the document]
Comment (NareshGopal) [reply manager/gui/src/components/alerts/LrAlert.constant.ts]: The default icon has been updated.

File: manager/gui/src/components/alerts/LrAlertComments.vue [hunk 2]
```diff
@@ -76,22 +111,22 @@
...
+  @Prop({ required: false, default: () => [] })
+  private alertComments: AlertCommentModel[];
   @PropSync("showAlertCommentsDialog", { required: false, default: false })
   private dialog: boolean;
   private commentText = "";
-  private alertComments: AlertCommentModel[] = [];
   private errorMsg = "";
 
-  mounted() {
-    Api.getData("alerts/comment", { isDummy: true }).then((resp: any) => {
```
[the remark this answers is not in the document]
Comment (NareshGopal) [reply manager/gui/src/components/alerts/LrAlertComments.vue:92]: Comments are now got with a separate API call with the Alert Id.

**Review by sujit-h-singh — APPROVED** [review]

File: manager/gui/src/components/alerts/LrAlertCommentReplies.vue [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+        <div class="comment-text mt-1">
+          {{ reply.comment_text }}
+        </div>
+        <div class="mt-2 comment-timestamp">
+          {{ formattedTime(new Date(reply.created_time * 1000)) }}
+        </div>
+      </li>
+    </ul>
+    <div class="add-reply">
+      <v-text-field
```
Comment (sujit-h-singh) [inline manager/gui/src/components/alerts/LrAlertCommentReplies.vue:47]: this is not as per our design, label should not be there.

File: manager/gui/src/components/alerts/LrAlertCommentReplies.vue [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+* GNU Affero General Public License for more details.
+* You should have received a copy of the GNU Affero General Public License
+* along with this program. If not, see <https://www.gnu.org/licenses/>.
+* For any questions about this software or licensing,
+* please email opensource@seagate.com.
+-->
+<template>
+  <div>
+    <span
+      class="expand-collapse"
```
Comment (sujit-h-singh) [inline manager/gui/src/components/alerts/LrAlertCommentReplies.vue:20]: design change as per discussion (color)

File: manager/gui/src/components/alerts/LrAlert.vue [hunk 5]
```diff
@@ -100,35 +107,91 @@ export default class LrAlert extends Vue {
...
+  }
+
+  occurrencesHandler(selectedRow: any) {
+    this.$router.push({
+      name: "alert-details",
+      params: { alertId: selectedRow.alert_uuid },
+    });
+  }
+
+  @Watch("$route")
```
Comment (sujit-h-singh) [inline manager/gui/src/components/alerts/LrAlert.vue:191]: why is this required?

File: manager/gui/src/components/alerts/LrAlert.vue [hunk 6]
```diff
@@ -25,8 +25,11 @@
       :multiSelectButtons="alertConst.alertTable.multiSelectButtons"
```
Comment (sujit-h-singh) [inline manager/gui/src/components/alerts/LrAlert.vue:25]: Other table design changes, column width and checkboxes.

File: manager/gui/src/router/index.ts [hunk 7]
```diff
@@ -57,6 +57,12 @@ const routes: Array<RouteConfig> = [
     component: LrAlert,
     props: true,
   },
+  {
```
Comment (sujit-h-singh) [inline manager/gui/src/router/index.ts:60]: this is not required

**Review by sujit-h-singh — CHANGES_REQUESTED** [review]

File: manager/gui/src/components/alerts/LrAlertCommentReplies.vue [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+        :key="reply.comment_id"
+      >
+        <div class="created-by">
+          {{ reply.created_by }}
+        </div>
+        <div class="comment-text mt-1">
+          {{ reply.comment_text }}
+        </div>
+        <div class="mt-2 comment-timestamp">
+          {{ formattedTime(new Date(reply.created_time * 1000)) }}
```
[the remark this answers is not in the document]
Comment (NareshGopal) [reply manager/gui/src/components/alerts/LrAlertCommentReplies.vue]: Done.

File: manager/gui/src/components/alerts/LrAlertCommentReplies.vue (hunk 4 above)
Comment (NareshGopal) [reply manager/gui/src/components/alerts/LrAlertCommentReplies.vue:20]: Seagate primary color has been applied to that text.

File: manager/gui/src/router/index.ts (hunk 7 above)
Comment (NareshGopal) [reply manager/gui/src/router/index.ts]: Removed the extra route for Alerts page.

File: manager/gui/src/components/alerts/LrAlert.vue (hunk 5 above)
Comment (NareshGopal) [reply manager/gui/src/components/alerts/LrAlert.vue]: Removed the watch for the route.

File: manager/gui/src/components/alerts/LrAlert.vue (hunk 6 above)
Comment (NareshGopal) [reply manager/gui/src/components/alerts/LrAlert.vue:25]: Done

File: manager/gui/src/components/alerts/LrAlertCommentReplies.vue (hunk 3 above)
Comment (NareshGopal) [reply manager/gui/src/components/alerts/LrAlertCommentReplies.vue:47]: Replaced the label with the placeholder.

File: manager/gui/src/components/alerts/LrAlert.vue [hunk 9]
```diff
@@ -70,25 +73,31 @@ import {
...
   showAlertDetailsDialog = false;
   selectedRecord: any = null;
   showAlertCommentsDialog = false;
+  public acknowledgeModal = create<SgtDialogModel>(SgtDialog);
 
   mounted() {
-    Api.getData("alerts/list", { isDummy: true }).then((resp: any) => {
-      this.alerts = resp["list"];
-    });
+    this.getAlertsList();
```
Comment (sujit-h-singh) [inline manager/gui/src/components/alerts/LrAlert.vue:97]: this should be in else part, since in case of severity it is making 2 api calls

File: manager/gui/src/components/alerts/LrAlert.vue [hunk 10]
```diff
@@ -100,35 +109,94 @@ export default class LrAlert extends Vue {
...
   }
+
   updateRecord(tableDataConfig: SgtDataTableFilterSortPag) {
-    // code for API call
+    let queryFilters = ``;
+    tableDataConfig.filterList.forEach((filter) => {
+      queryFilters += `${filter.name}=${filter.value}`;
+    });
+    this.getAlertsList(queryFilters);
     this.chips = tableDataConfig.filterList;
```
Comment (sujit-h-singh) [inline manager/gui/src/components/alerts/LrAlert.vue:150]: chips should be updated only after API returns value

File: manager/gui/src/components/alerts/LrAlert.constant.ts [hunk 11]
```diff
@@ -87,7 +88,12 @@ export const lrAlertConst = {
         align: "middle",
       },
       { text: "State", value: "state", width: "125px" },
-      { text: "Description", value: "description", type: "custom" },
+      {
+        text: "Description",
+        value: "description",
+        type: "custom",
+        width: "200px",
```
Comment (sujit-h-singh) [inline manager/gui/src/components/alerts/LrAlert.constant.ts:95]: you need to increase width for description column.

File: manager/gui/src/components/alerts/LrAlert.vue [hunk 12]
```diff
@@ -100,35 +107,91 @@ export default class LrAlert extends Vue {
...
+      });
+      this.alertConst.searchConfig.advanceForm = updatedAdvanceForm;
+    }
+  }
+
   getColor(item: any) {
     return this.alertConst.severityList[item.severity];
   }
+
   updateRecord(tableDataConfig: SgtDataTableFilterSortPag) {
```
[the remark this answers is not in the document]
Comment (sujit-h-singh) [reply manager/gui/src/components/alerts/LrAlert.vue:144]: Was related to other filters.

File: manager/gui/src/components/alerts/LrAlertCommentReplies.vue [hunk 13]
```diff
@@ -0,0 +1,117 @@
+<!--
```
Comment (sujit-h-singh) [inline manager/gui/src/components/alerts/LrAlertCommentReplies.vue:1]: there are multiple console errors and warning while opening show more replies.

File: manager/gui/src/components/alerts/LrAlertCommentReplies.vue [hunk 14]
```diff
@@ -0,0 +1,117 @@
...
+  name: "LrAlertCommentReplies",
+  components: {},
+})
+export default class LrAlertCommentReplies extends Vue {
+  isRepliesVisible = false;
+  @Prop({ required: true }) private comment_id: string;
+  @Prop({ required: true }) private replies: IAlertComment[];
+
+  formattedTime(time: number) {
+    return formatTime(`${new Date(time * 1000)}`).replace(" ", " | ");
```
Comment (sujit-h-singh) [inline manager/gui/src/components/alerts/LrAlertCommentReplies.vue:78]: can this be a util function, to be used at other places.

File: manager/gui/src/lib/components/SgtDataTable/SgtDataTable.vue [hunk 15]
```diff
@@ -389,12 +391,13 @@ export default class SgtDataTable extends Vue {
...
   updateFilterByChip(chip: SgtFilterObject) {
-    const ind = this.tableDataConfig.filterList.findIndex(
+    const formField = this.searchConfig.advanceForm?.find(
       (ele) => ele.name == chip.name
     );
-    if (ind >= 0) {
-      this.tableDataConfig.filterList.splice(ind, 1);
-      this.updateRecord();
+    if (formField) {
+      formField.value = "";
```
Comment (sujit-h-singh) [inline manager/gui/src/lib/components/SgtDataTable/SgtDataTable.vue:398]: Make this null or undefined, will handle cases for type number, checkboxes, dropdowns, radios, etc.

Comment (NareshGopal) [reply manager/gui/src/lib/components/SgtDataTable/SgtDataTable.vue]: Assigned 'null' to this variable.

File: manager/gui/src/components/alerts/LrAlertCommentReplies.vue (hunk 14 above)
Comment (NareshGopal) [reply manager/gui/src/components/alerts/LrAlertCommentReplies.vue]: Created new utility function "formatCommentTime".

File: manager/gui/src/components/alerts/LrAlertCommentReplies.vue (hunk 13 above)
Comment (NareshGopal) [reply manager/gui/src/components/alerts/LrAlertCommentReplies.vue:1]: It was due to the duplicate keys. Modified the Ids in the dummy data.

File: manager/gui/src/components/alerts/LrAlert.vue (hunk 9 above)
Comment (NareshGopal) [reply manager/gui/src/components/alerts/LrAlert.vue]: Moved the getAlertsList function call to the else part.

File: manager/gui/src/components/alerts/LrAlert.vue (hunk 10 above)
Comment (NareshGopal) [reply manager/gui/src/components/alerts/LrAlert.vue]: Moved the chips assignment after the API call.

File: manager/gui/src/components/alerts/LrAlert.constant.ts (hunk 11 above)
Comment (NareshGopal) [reply manager/gui/src/components/alerts/LrAlert.constant.ts]: Added a fixed column width to the action column, so that the description column can expand to the maximum available width in the table.

File: manager/gui/src/components/alerts/LrAlert.vue (hunk 12 above)
[the remark this answers is not in the document]
Comment (NareshGopal) [reply manager/gui/src/components/alerts/LrAlert.vue:147]: The filter feature is applied to the 'severity' column in the table in the advance search form.

**Review by sujit-h-singh — APPROVED** [review]

**Review by sujit-h-singh — CHANGES_REQUESTED** [review]
Tests are pending for table headers, advance search fields, fields validation, search button click, alert severity, is multiselect present,

**Review by sujit-h-singh — APPROVED** [review]
